src/domain/inventory/locations/new/index.tsx

```diff
@@ -0,0 +1,192 @@
+import {
+  AdminPostStockLocationsReq,
+  SalesChannel,
+  StockLocationAddressDTO,
+  StockLocationAddressInput,
+} from "@medusajs/medusa"
+import {
+  useAdminAddLocationToSalesChannel,
+  useAdminCreateStockLocation,
+} from "medusa-react"
+import { useForm } from "react-hook-form"
+import Button from "../../../../components/fundamentals/button"
+import CrossIcon from "../../../../components/fundamentals/icons/cross-icon"
+import FocusModal from "../../../../components/molecules/modal/focus-modal"
+import Accordion from "../../../../components/organisms/accordion"
+import { useFeatureFlag } from "../../../../context/feature-flag"
+import useNotification from "../../../../hooks/use-notification"
+import { getErrorMessage } from "../../../../utils/error-messages"
+import { nestedForm } from "../../../../utils/nested-form"
+import AddressForm from "../components/address-form"
+import GeneralForm, { GeneralFormType } from "../components/general-form"
+import SalesChannelsForm from "../components/sales-channels-form"
+
+type NewLocationForm = {
+  general: GeneralFormType
+  address: StockLocationAddressDTO
+  salesChannels: {
+    channels: SalesChannel[]
+  }
+}
+
+const NewLocation = ({
+  onClose,
+  refetch,
+}: {
+  onClose: () => void
+  refetch: () => void
+}) => {
+  const form = useForm<NewLocationForm>({
+    defaultValues: {
+      general: {
+        name: "",
+      },
+      address: undefined,
+      salesChannels: {
+        channels: [],
+      },
+    },
+    reValidateMode: "onBlur",
+    mode: "onBlur",
+  })
+  const { handleSubmit, formState } = form
+
+  const notification = useNotification()
+  const { isFeatureEnabled } = useFeatureFlag()
+
+  const { mutate } = useAdminCreateStockLocation()
+  const { mutateAsync: associateSalesChannel } =
+    useAdminAddLocationToSalesChannel()
+
+  const createSalesChannelAssociationPromise = (salesChannelId, locationId) =>
+    associateSalesChannel({
+      id: salesChannelId,
+      payload: { location_id: locationId },
+    })
+
+  const onSubmit = () =>
+    handleSubmit(async (data) => {
+      const { locationPayload, salesChannelsPayload } = createPayload(data)
+      mutate(locationPayload, {
+        onSuccess: ({ stock_location }) => {
+          Promise.all(
+            salesChannelsPayload.map((salesChannel) =>
+              createSalesChannelAssociationPromise(
+                salesChannel.id,
+                stock_location.id
+              )
+            )
+          )
+            .then(() => {
+              notification("Success", "Location added successfully", "success")
+            })
+            .catch(() => {
+              notification(
+                "Error",
+                "Location was created successfully, but there was an error associating sales channels",
+                "error"
+              )
+            })
+            .finally(() => {
+              refetch()
```

Comment:
question: what are we explicitly refetching, could this be handled with cache invalidation and autorefetch?

Reply:
Nothing needed anymore, will remove this in line with a comment from Kasper above.